src/ripple/overlay/impl/OverlayImpl.h

```diff
@@ -456,6 +456,60 @@ class OverlayImpl : public Overlay
 
     void
     sendEndpoints();
+
+private:
+    struct Stats
+    {
+
+        template <class Handler>
+        Stats (Handler const& handler, beast::insight::Collector::ptr const& collector)
+            : hook (collector->make_hook (handler))
+            , peerDisconnects (collector->make_gauge("Overlay","Peer_Disconnects")) 
+            { 
+                mcollector=collector;
+            }
+
+        beast::insight::Hook hook;
+        beast::insight::Gauge peerDisconnects;
+        std::vector<beast::insight::Gauge> trafficGauges;
+        beast::insight::Collector::ptr mcollector;
+        
+    };
+    
+    Stats m_stats;
+
+private:
+    void collect_metrics()
+    {   
+        std::lock_guard lock (mutex_);
+
+        auto const stats = m_traffic.getCounts();
+
+        if(m_stats.trafficGauges.empty()){
+            for (auto const& i : stats)
```

Comment:
This variable is called `i`. It's not an integer, which is what most people expect an `i` to be. There is an integer named `i` declared just after this block; two variables with the same name very close in scope but with different types is inviting confusion. The same variable in the same iteration pattern just below is named `s`; can we name this one `s` too?

Reply:
Thanks, I made these changes.